src/main/java/org/jabref/logic/journals/PredatoryJournalLoader.java

```diff
@@ -0,0 +1,140 @@
+package org.jabref.logic.journals;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Pattern;
+
+import org.jabref.logic.net.URLDownload;
+
+import kong.unirest.UnirestException;
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVParser;
+import org.apache.commons.csv.CSVRecord;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class PredatoryJournalLoader {
+    private static class PJSource {
+        URL URL = null;
+        final String ELEMENT_REGEX;
+
+        PJSource(String URL, String ELEMENT_REGEX) {
+            try {
+                this.URL = new URL(URL);
+            } catch (MalformedURLException ex) {
+                PredatoryJournalLoader.logException(ex);
+            }
+            this.ELEMENT_REGEX = ELEMENT_REGEX;
+        }
+    }
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(PredatoryJournalLoader.class);
+    private static final List<PJSource> PREDATORY_SOURCES = List.of(
+        new PJSource("https://raw.githubusercontent.com/stop-predatory-journals/stop-predatory-journals.github.io/master/_data/journals.csv",
+                    null),
+        /*
+        new PJSource("https://raw.githubusercontent.com/stop-predatory-journals/stop-predatory-journals.github.io/master/_data/hijacked.csv",
+                    null),
+        */
+        new PJSource("https://raw.githubusercontent.com/stop-predatory-journals/stop-predatory-journals.github.io/master/_data/publishers.csv",
+                    null),
+        new PJSource("https://beallslist.net/",
+                    "<li>.*?</li>"),
+        new PJSource("https://beallslist.net/standalone-journals/",
+                    "<li>.*?</li>"),
+        new PJSource("https://beallslist.net/hijacked-journals/",
+                    "<tr>.*?</tr>")
+    );
+    private static PredatoryJournalRepository repository;
+    private static List<String> linkElements = new ArrayList<>();
+
+    public static PredatoryJournalRepository loadRepository(boolean doUpdate) {
```

Comment:
Do not mix update with loading. Please remove the boolean parameter and never (!) update the list.